### PR TITLE
ADM-662: [frontend] add version in header

### PR DESCRIPTION
**Pull request by neomgb**
## Summary

add version in header

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.
image

## After

_Description_

**Screenshots**
image

## Note

_Null_

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -31,7 +31,11 @@ body:
   id: version
   attributes:
     label: Version
-    description: What version of our software are you running? You can get the version by add "api/v1/version" after the host address you visit the heartbeat. If you can't get version by the url, you can input "1.0.0" as well
+    description: |
+      What version of our software are you running? There is two methods to get the current version:
+        1. you can see the version info at the top of the heartbeat pages
+        2. Add “api/v1/version”after the host address you visit the heartbeat
```
Comment (Zxx-Ivy) [inline .github/ISSUE_TEMPLATE/bug_report.yml:37]: Please use English quotation marks

File: frontend/__tests__/src/layouts/Header.test.tsx [hunk 2]
```diff
@@ -15,6 +24,34 @@ describe('Header', () => {
     expect(getByText(PROJECT_NAME)).toBeInTheDocument()
   })
 
+  it('should show version info when request succeed', async () => {
+    headerClient.getVersion = jest.fn().mockResolvedValueOnce('1.11')
+    let component: any
```
Comment (Zxx-Ivy) [inline frontend/__tests__/src/layouts/Header.test.tsx:29]: Please fix issues of github actions

File: frontend/__tests__/src/layouts/Header.test.tsx [hunk 3]
```diff
@@ -15,6 +24,34 @@ describe('Header', () => {
...
+        </BrowserRouter>
+      )
+    })
+    const element = component.getByText(/v1.11/)
+    expect(element).toBeInTheDocument()
+  })
+
+  it('should show version info when request failed', async () => {
+    headerClient.getVersion = jest.fn().mockResolvedValueOnce('')
+    let component: any
```
Comment (Zxx-Ivy) [inline frontend/__tests__/src/layouts/Header.test.tsx:43]: the same

File: frontend/__tests__/src/layouts/Header.test.tsx [hunk 4]
```diff
@@ -15,6 +24,34 @@ describe('Header', () => {
     expect(getByText(PROJECT_NAME)).toBeInTheDocument()
   })
 
+  it('should show version info when request succeed', async () => {
+    headerClient.getVersion = jest.fn().mockResolvedValueOnce('1.11')
+    let component: any
+    await act(async () => {
+      component = render(
+        <BrowserRouter>
```
Comment (Zxx-Ivy) [inline frontend/__tests__/src/layouts/Header.test.tsx:32]: Maybe we can extract a function for those piece of lines

File: frontend/__tests__/src/layouts/Header.test.tsx [hunk 5]
```diff
@@ -15,6 +24,34 @@ describe('Header', () => {
...
+    })
+    const element = component.getByText(/v1.11/)
+    expect(element).toBeInTheDocument()
+  })
+
+  it('should show version info when request failed', async () => {
+    headerClient.getVersion = jest.fn().mockResolvedValueOnce('')
+    let component: any
+    await act(async () => {
+      component = render(
```
Comment (Zxx-Ivy) [inline frontend/__tests__/src/layouts/Header.test.tsx:45]: Maybe we can extract a function for those piece of lines

File: frontend/cypress/pages/home.ts [hunk 6]
```diff
@@ -11,6 +11,10 @@ class Home {
     this.createANewProjectButton().click()
   }
 
+  checkVersion = () => {
+    cy.get('span[title="Heartbeat"]').parent().next().should('exist')
```
Comment (Zxx-Ivy) [inline frontend/cypress/pages/home.ts:15]: Please follow the POM pattern and avoid using assert at the page level

File: frontend/README.md [hunk 7]
```diff
@@ -33,11 +63,10 @@ export const StyledTypography = styled(Typography)({
...
+2. css units should use rem:
 
 ```
-export const StyledTypography = styled(Typography)({
-  // fontSize: '0.88rem',
-  fontSize: '14px',
+export const StyledTypography = styled('div')({
+  width: '10rem',
 })
 ```
```
Comment (Zxx-Ivy) [inline frontend/README.md:72]: Maybe we can add a new one, e2e requires the use of POM's model

File: frontend/__tests__/src/layouts/Header.test.tsx [hunk 8]
```diff
@@ -1,10 +1,28 @@
...
+  beforeEach(() => {
+    headerClient.getVersion = jest.fn().mockResolvedValue('')
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  const setup = () =>
+    act(async () =>
```
Comment (Zxx-Ivy) [inline frontend/__tests__/src/layouts/Header.test.tsx:18]: Need to put "act" to assert block

**Review by Zxx-Ivy — APPROVED** [review]

File: frontend/__tests__/src/layouts/Header.test.tsx [hunk 9]
```diff
@@ -1,38 +1,56 @@
...
     )
 
+  it('should show project name', () => {
+    const { getByText } = setup()
+
     expect(getByText(PROJECT_NAME)).toBeInTheDocument()
   })
 
+  it('should show version info when request succeed', async () => {
+    headerClient.getVersion = jest.fn().mockResolvedValueOnce('1.11')
```
Comment (guzhongren) [inline frontend/__tests__/src/layouts/Header.test.tsx:31]: @neomgb using `1.11` from fixtures.ts

File: frontend/__tests__/src/client/HeaderClient.test.ts [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+const server = setupServer(rest.get(MOCK_VERSION_URL, (req, res, ctx) => res(ctx.status(HttpStatusCode.Ok))))
+
+describe('header client', () => {
+  beforeAll(() => server.listen())
+  afterEach(() => server.resetHandlers())
+  afterAll(() => server.close())
+
+  it('should get response when get header status 200', async () => {
+    const excepted = '1.11'
```
Comment (guzhongren) [inline frontend/__tests__/src/client/HeaderClient.test.ts:15]: fixturet.ts

File: frontend/__tests__/src/layouts/Header.test.tsx [hunk 11]
```diff
@@ -1,38 +1,56 @@
...
+    const { getByText } = await act(async () => setup())
+
+    expect(getByText(/v1.11/)).toBeInTheDocument()
```
Comment (guzhongren) [inline frontend/__tests__/src/layouts/Header.test.tsx:34]: fixtures.ts

**Review by guzhongren — COMMENTED** [review]
PLZ update commit with the comments

suggestion: PLZ refactor code after coding. Ref to `red green blue` of the refactor

**Pull request merged**